src/model/mongodb/mongodb-repository.ts

```diff
@@ -128,7 +128,7 @@ class MongoDBArticleRepository implements ArticleRepository {
     }));
   }
 
-  async getArticleVersion(identifier: string): Promise<EnhancedArticleWithVersions> {
+  async getArticleVersion(identifier: string): Promise<EnhancedArticleWithVersions | null> {
```

Comment:
Small-is nitpick: Can we change the API name to `findArticleVersion` so to tie up the semantics of get* (in which asking for something that is not there is an error) and find* (in which you expect there to potentially be nothing return)